### DISCUSSION TITLE
py mammotion in other platforms?

### DISCUSSION THREAD

**Discussion by nieand1985** [mikey0000/PyMammotion#68] [Ideas]
Hello,

are there any plans to integrate pymammtion into io broker or other platforms? 

many thanks in advance

**Comment by mikey0000** [comment]
Sorry I just do not have the capacity to do this. The best possible solution is to use a MQTT bridge. Io broker would require a javascript port of this library (from what I can see, first time hearing of io broker) , I definitely don't have capacity for that. Anyone is welcome to use this library for other platforms, its full featured enough to use without the home assistant side.

Porting it is definitely doable but probably worth waiting till it's more stable.

**Comment by ESF-nl** [comment]
I was wondering what adjustments are required for Domoticz? This home automation runs on a Raspberry Pi and has MQTT running natively. Can I just follow the steps as your Home Automation?

**Reply by mikey0000** [reply]
As per above an MQTT bridge, pymammotion is written in python, no idea how you'd use that with a project written in lua.

**Reply by ESF-nl** [reply]
Maybe together we can make your great project work in Domoticz as well. I've installed pymammotion with pip3. In my MQTT I've made a topic 'lawn'. 1) How to run the script to publish? 2) In what file are the settings for the right IP address, port and MQTT topic?

**Reply by mikey0000** [reply]
I'll write some instructions for you when I get a moment, you'll need to hook into the MQTT callbacks and re-publish the topics from the mammotion MQTT servers, I say servers as new models use a different MQTT server to older models

**Reply by mikey0000** [reply]
Here is a basic example of how to hook into the MQTT events, there are two MQTT services now due to newer Mammotion mowers using a different MQTT service so you need to hook into both the aliyun and mammotion MQTT services if your mower uses them, so you will need to check for the existence of the mqtt service, e.g a yuka mini uses the old aliyun mqtt, the vision model uses mammotion, so if you only had a vision you'd only have the mammotion mqtt 

```
                  await self.mammotion.login_and_initiate_cloud(
                        credentials["email"], credentials["password"]
                    )
                    # Get cloud_client after login

                mqtt = self.mammotion.mqtt_list[f"{credentials["email"]}_aliyun"]
                mqtt_mammotion = self.mammotion.mqtt_list[f"{credentials["email"]}_mammotion"]

                # mqtt.mqtt_message_event.add_subscribers(self._on_mqtt_message)

                mqtt.mqtt_device_event.add_subscribers(self._on_device_event)

                mqtt.mqtt_status_event.add_subscribers(self._on_status)

                mqtt.mqtt_properties_event.add_subscribers(self._on_properties)

```

**Reply by mikey0000** [reply]
On top of this there is a mower dataclass which records the state of the mower, you can look at this state as often as you want and might want to instead just publish the state of the mower. I recommend looking at the home assistant api py file as its a layer to manage calling the MQTT api etc and all you'd need to do is call the update function and then just read the state of the mower.

**Reply by svefro** [reply]
There is mammotion_to_mqtt.py that can be used to publish all parameters to a mqtt broker. Take a look at that.
I'm using this to do everything. You can also send commands over mqtt to pymammotion.

You need to handle all the data your self.

I'm running it inside a self contained docker container using python:3.14.2-slim as a base image.
So you do not even need to mess with python on your os.

Just be aware that you are not spamming commands to it as it might get your account blocked if you are not careful.
